doc/tntcxx_getting_started.rst

```diff
@@ -0,0 +1,524 @@
+the network provider as template parameters (note that they must be the same
+as ones of the client):
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel04-1
+   :end-before: doclabel04-2
+   :language: c
+
+.. _gs_cxx_connect_connecting:
+
+Connecting
+~~~~~~~~~~
+
+Our :ref:`Tarantool instance <gs_cxx_prereq_tnt_run>` is listening to
+the ``3301`` port on ``localhost``.
+Let's define the corresponding variables as well as the ``WAIT_TIMEOUT`` variable
+for connection timeout.
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel05-1
+   :end-before: doclabel05-2
+   :language: c
+
+To connect to the Tarantool instance, we should invoke
+the ``Connector::connect()`` method of the client object and
+pass three arguments: connection instance, address, and port.
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel06-1
+   :end-before: doclabel06-2
+   :language: c
+
+.. _gs_cxx_connect_error:
+
+Error handling
+~~~~~~~~~~~~~~
+
+Implementation of the connector is exception free, so we rely on the return
+codes: in case of fail, the ``connect()`` method returns ``rc < 0``. To get the
+error message corresponding to the last error occured during
+communication with the instance, we can invoke the ``Connection::getError()``
+method.
+
+.. literalinclude:: ./_includes/Simple.cpp
+   :start-after: doclabel06-2
+   :end-before: doclabel06-3
+   :language: c
+
+To reset connection after errors, that is, to clean up the error message and
+connection status, the ``Connection::reset()`` method is used.
+
+.. // TBD For the detailed information on the connector's API, refer to <link to the API document parts related to the API used here> -- implement this line when API document is ready.
+
+.. _gs_cxx_requests:
+
+Working with requests
+---------------------
+
+In this section, we will show how to:
+
+* :ref:`prepare different types of requests <gs_cxx_requests_prepare>`
+* :ref:`send the requests <gs_cxx_requests_send>`
+* :ref:`receive and handle responses <gs_cxx_requests_response>`.
+
+We will also go through the :ref:`case of having several connections <gs_cxx_requests_several>`
+and executing a number of requests from different connections simultaneously.
+
+In our example C++ application, we execute the following types of requests:
+
+* ping
+* replace
+* select.
+
+.. NOTE::
+
+   Examples on other request types, namely, ``insert``, ``delete``, ``upsert``,
+   and ``update``, will be added to this manual later.
+
+Each request method returns a request ID that is a sort of `future <https://en.wikipedia.org/wiki/Futures_and_promises>`_.
+It can be used to get the result of request execution once it is ready, that is,
+response. Requests are queued in the output buffer of connection
+until the ``Connector::wait()`` method is called.
+
+.. _gs_cxx_requests_prepare:
+
+Preparing requests
+~~~~~~~~~~~~~~~~~~~
+
+At this step, requests are encoded in the msgpack format and saved in the
```

Comment:
I believe, it's called [MessagePack](https://msgpack.org/)
```suggestion
At this step, requests are encoded in the MessagePack format and saved in the
```

Reply:
Corrected.